### PR TITLE
Fix validate -u HTTP 400 error by removing deprecated pds:Resource registry query filters

### LINKED ISSUE #1657: validate -u fails: `ERROR [error.connection.registry] Error connecting to Registry`
### Checked for duplicates

No - I haven't checked

### 🐛 Describe the bug

validate -u ERRORs out instead of updating the resource file

### 🕵️ Expected behavior

I expected [...]

### 📜 To Reproduce

From my mac:
% validate -u
[snip]
  FAIL: file:/Users/rchen/PDS4tools/validate/resources/registered_context_products.json
      ERROR  [error.connection.registry]   Error connecting to Registry to update registered context products config file. Verify internet connection and try again.
        1 product validation(s) completed
java.io.IOException: Server returned HTTP response code: 400 for URL: https://pds.mcp.nasa.gov/api/search/1/products?limit=1000&sort=ops:Harvest_Info.ops:harvest_date_time&q=%28product_class+eq+%22Product_Context%22+and+pds%3AResource.pds%3Atype+ne+%22Information.Science_Portal%22+and+pds%3AResource.pds%3Atype+ne+%22Information.Investigation%22%29
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:2014)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1615)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:223)
	at java.base/java.net.URL.openStream(URL.java:1325)
	at gov.nasa.pds.validate.ValidateLauncher.getLatestJsonContext(ValidateLauncher.java:552)
	at gov.nasa.pds.validate.ValidateLauncher.processMain(ValidateLauncher.java:1809)
	at gov.nasa.pds.validate.ValidateLauncher.main(ValidateLauncher.java:1888)
java.lang.NullPointerException: Cannot invoke "java.util.Scanner.close()" because "reader" is null

### 🖥 Environment Info

%validate --version
gov.nasa.pds:validate
Version 4.2.0
Release Date: 2026-06-15 22:25:06

I'm pretty confident the ingest to OpenSearch worked. I added 2 new LIDs, and here's one:
(base) [pds4@cds-prod-gamma ~]$ LIDVID=urn:nasa:pds:context:target:asteroid.162421_2000_et70::1.0
(base) [pds4@cds-prod-gamma ~]$ curl --get "https://pds.mcp.nasa.gov/api/search/1/products/$LIDVID"
{"id":"urn:nasa:pds:context:target:asteroid.162421_2000_et70::1.0","type":"Product_Context","title":"(162421) 2000 ET70","investigations":[],"observing_system_components":[],"targets":[], [snip]

although the ingest log had errors I did not understand:
(base) [pds4@cds-prod-gamma ~]$ grep ERROR log/harvest.context_pds4-20260819110135.log 
2026-08-19 11:01:55 [ERROR] (LddEsJsonWriter.java:writeRecord:141) The field 'directory_path_name' will not be searchable because LDD does not contain a type for it (see harvest#204).
2026-08-19 11:01:58 [ERROR] (SchemaUpdater.java:updateSchema:131) Could not find the data type for the field pds:Product_Context/pds:Reference_List
2026-08-19 11:01:58 [ERROR] (FilesProcessor.java:onFile:172) Could not find datatype(s). See 'https://nasa-pds.github.io/registry/user/load1.html' for more information.
2026-08-19 11:02:02 [ERROR] (SchemaUpdater.java:updateSchema:131) Could not find the data type for the field pds:Product_Context/pds:Discipline_Area
2026-08-19 11:02:02 [ERROR] (FilesProcessor.java:onFile:172) Could not find datatype(s). See 'https://nasa-pds.github.io/registry/user/load1.html' for more information.
[snip][

### 📚 Version of Software Used

_No response_

### 🩺 Test Data / Additional context

_No response_

### 🦄 Related requirements

🦄 #xyz

### ⚙️ Engineering Details

_No response_

### 🎉 Integration & Test

_No response_

### LINKED ISSUE #1659: Add automated test coverage for validate -u registry update (getLatestJsonContext)
## Summary

Add automated test coverage for the `validate -u` flag / `getLatestJsonContext()` method in `ValidateLauncher`. Currently there are zero tests for this code path.

The method makes a paginated HTTP call to the PDS Search API to download registered context products and write them to a local JSON file. A mock HTTP server (e.g. WireMock or a simple `com.sun.net.httpserver.HttpServer`) should be stood up in the test to serve a known JSON response, and the test should assert:

1. The output JSON file is written successfully
2. The correct number of context products is reported
3. Pagination (`search-after`) is handled correctly when total > page size
4. An HTTP error (e.g. 400, 500) produces `error.connection.registry` in the report rather than an uncaught exception

## Related Issues

Refs #1657
Refs #1658

---

### PR DESCRIPTION
**Pull request by jordanpadams**
## 🗒️ Summary

Remove `pds:Resource.pds:type` filter clauses from the registry search query used by `validate -u` — these resource fields no longer exist in the registry and cause HTTP 400 errors. Also updates the pagination sort key from `ops:Harvest_Info.ops:harvest_date_time` to `ops:Label_File_Info.ops:creation_date_time`.

Additionally fixes a pre-existing `NullPointerException` in the `finally` block of `getLatestJsonContext()` when the HTTP connection fails before the `Scanner` is assigned.

Changed files:
- `src/main/resources/validate.properties` — simplified query to `(product_class eq "Product_Context")`
- `src/main/java/gov/nasa/pds/validate/ValidateLauncher.java` — updated `searchAfterKey`; fix NPE in `finally` block; make `getLatestJsonContext()` package-private for testability
- `src/main/java/gov/nasa/pds/validate/util/ToolInfo.java` — add `setProperty()` to support overriding search URL in tests
- `src/test/java/gov/nasa/pds/validate/GetLatestJsonContextTest.java` — new: 3 tests using a JDK `HttpServer` mock (happy path, pagination, HTTP 400 error)

### 🤖 AI Assistance Disclosure
- [ ] No AI assistance used
- [ ] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [ ] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [x] AI generated substantial portions of this code

Estimated % of code influenced by AI: 90%

## ⚙️ Test Data and/or Report

Automated tests added in `GetLatestJsonContextTest` — 3 scenarios using a local JDK `HttpServer` mock (no network required):

1. **Happy path** — single-page response with 2 products; asserts output JSON written with correct `Product_Context` array
2. **Pagination** — total=3 served over 2 pages; asserts exactly 2 HTTP requests made
3. **HTTP 400 error** — asserts method does not throw and output file is not written

All 3 tests pass locally (`mvn test -Dtest=GetLatestJsonContextTest`).

## ♻️ Related Issues

Fixes #1657
Fixes #1659

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

---
🤖 Generated with [Claude Code](https://claude.ai/claude-code)

### CODE REVIEW

File: src/main/resources/validate.properties [hunk 1]
```diff
@@ -37,4 +37,4 @@ validate.copyright=\nCopyright 2019, by the California Institute of Technology (
 validate.search_url=https://pds.nasa.gov/api
 validate.output_file_name=registered_context_products.json
 validate.endpoint=search/1/products
-validate.query=(product_class eq "Product_Context" and pds:Resource.pds:type ne "Information.Science_Portal" and pds:Resource.pds:type ne "Information.Investigation")
+validate.query=(product_class eq "Product_Context")
```
Comment (jordanpadams) [inline src/main/resources/validate.properties:40]: This fix is because we no longer load these resources into the Registry, which was causing these queries to fail.

**Review by nutjob4life — APPROVED** [review]
Fantastic @jordanpadams, thank you so much.

With the help of an artificial intelligence, we've found a potential pagination issue:

The `creation_date_time` may not be unique in the registry and using `search-after` may skip records. For example, suppose the page size is 1000 and we see records like:
```
…
record  999   creation_date_time = 2024-01-01T12:00:00Z
record 1000   creation_date_time = 2024-01-01T12:00:00Z
record 1001   creation_date_time = 2024-01-01T12:00:00Z
record 1002   creation_date_time = 2024-01-01T12:00:00Z
```
If page 1 ends at record №1000 and the next request says

    search-after=2024-01-01T12:00:00Z

what happens to record №1001 and №1002? It might be that the underlying API/OpenSearch has a deterministic tie-breaker; do we know?

Everything else looks copacetic though, and Maven's happy, which means: ✅

Maven details:
```
[INFO] Tests run: 326, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
```

**Comment by jordanpadams** [comment]
Great catch, @nutjob4life — the concern is valid. `ops:Label_File_Info.ops:creation_date_time` is not guaranteed to be unique, so a page boundary landing on a run of records with identical timestamps could cause `search-after` to skip some of them.

The good news: `lidvid` is already present in each record's `properties` (we already extract it in `parseJsonObjectWriteTofile`), and LIDVID is inherently unique. We can add it as a secondary sort key to guarantee a stable, unambiguous cursor.

The PDS Search API accepts multiple `sort` fields as a comma-separated list, so the fix in `getLatestJsonContext()` would be:

```java
final String searchAfterKey = "ops:Label_File_Info.ops:creation_date_time,lidvid";
```

and the `search-after` value becomes `<timestamp>,<lidvid>` — two values separated by a comma, consistent with how the API's multi-key search-after works.

I'll open a follow-up issue to track this so it doesn't get lost, and address it before the release cut.

> 🤖 Generated with [Claude Code](https://claude.ai/claude-code)
